crypto/objects/obj_dat.c

```diff
@@ -160,26 +198,33 @@ void ossl_obj_cleanup_int(void)
     lh_ADDED_OBJ_doall(added, cleanup3_doall); /* free objects */
     lh_ADDED_OBJ_free(added);
     added = NULL;
+    CRYPTO_THREAD_lock_free(ossl_obj_lock);
```

Comment:
This must be called even if added == NULL.

Reply:
Nice catch.  An earlier version allocated them together.